### PR TITLE
Add markdown contents for "A re-introduction to mkosi" blog post

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
Let's keep this in the repo so it can be reviewed and so we can update it later if needed.

### CODE REVIEW

File: blog/content/a-reintroduction-to-mkosi.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+I'll finish with a bunch of links to more information about `mkosi` and
+related tooling:
+
+- [Github repository](https://github.com/systemd/mkosi)
+- [Building RHEL and RHEL UBI images with mkosi](https://fedoramagazine.org/create-images-directly-from-rhel-and-rhel-ubi-package-using-mkosi/)
+- [My presentation on systemd-repart at ASG 2023](https://media.ccc.de/v/all-systems-go-2023-191-systemd-repart-building-discoverable-disk-images)
+- [mkosi's matrix channel](https://matrix.to/#/#mkosi:matrix.org).
+- [systemd's mkosi configuration](https://github.com/systemd/systemd)
+- [mkosi's mkosi configuration](https://github.com/systemd/mkosi)
```
Comment (behrmann) [inline blog/content/a-reintroduction-to-mkosi.md:324]: These two could be a bit more specific than the top level.

**Comment by DaanDeMeyer** [comment]
Well we're in a bit of a pickle there since until https://gitlab.com/virtio-fs/virtiofsd/-/merge_requests/207 is merged the `RuntimeTrees=` part of the virtiofsd integration is broken. Booting off virtiofs still works so I could mention that.

File: blog/content/a-reintroduction-to-mkosi.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+In the systemd mkosi configuration, we automatically use a tools tree if
+we detect your distribution does not have the minimum required systemd
+version installed.
+
+## Various other interesting features
+
+- Multiple images can be built at once by placing their configuration in
+  `mkosi.images/`. Dependencies between these images can be configured
+  with the `Dependencies=` setting. For example, you can first build a
+  custom initrd and then use it to build a bootable disk image
```
Comment (keszybz) [inline blog/content/a-reintroduction-to-mkosi.md:291]: Period.

File: blog/content/a-reintroduction-to-mkosi.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+- System extensions can be built using the `BaseTrees=` and `Overlay=`
+  settings. Point `BaseTrees=` to the base image, enable `Overlay=` and
+  set `Format=sysext` to have `mkosi` build an overlay on top of the
+  given base image and package it up as a system extension.
+- To sign any generated UKIs for secure boot, put your secure boot key
+  and certificate in `mkosi.key` and `mkosi.crt` and enable the
+  `SecureBoot=` setting. You can also run `mkosi genkey` to have `mkosi`
+  generate a key and certificate itself.
+- Profiles can be defined in the `mkosi.profiles/` directory. The
```
Comment (keszybz) [inline blog/content/a-reintroduction-to-mkosi.md:300]: This should be a separate paragraph.

File: blog/content/a-reintroduction-to-mkosi.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+we detect your distribution does not have the minimum required systemd
+version installed.
+
+## Various other interesting features
+
+- Multiple images can be built at once by placing their configuration in
+  `mkosi.images/`. Dependencies between these images can be configured
+  with the `Dependencies=` setting. For example, you can first build a
+  custom initrd and then use it to build a bootable disk image
+- System extensions can be built using the `BaseTrees=` and `Overlay=`
```
Comment (keszybz) [inline blog/content/a-reintroduction-to-mkosi.md:292]: This too…

File: blog/content/a-reintroduction-to-mkosi.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+- To sign any generated UKIs for secure boot, put your secure boot key
+  and certificate in `mkosi.key` and `mkosi.crt` and enable the
+  `SecureBoot=` setting. You can also run `mkosi genkey` to have `mkosi`
+  generate a key and certificate itself.
+- Profiles can be defined in the `mkosi.profiles/` directory. The
+  profile to use can be selected using the `Profile=` setting (or
+  `--profile=`) on the command line. Profiles allow selecting different
+  variants of the same image while still sharing most of the
+  configuration.
+- The `Ephemeral=` setting can be enabled to boot the image in an
```
Comment (keszybz) [inline blog/content/a-reintroduction-to-mkosi.md:305]: And this.

Comment (DaanDeMeyer) [reply blog/content/a-reintroduction-to-mkosi.md:369]: Does this really needs its own paragraph? `Ephemeral=` is pretty simple

Comment (keszybz) [reply blog/content/a-reintroduction-to-mkosi.md:369]: I think it does, i.e. it's a separate topic that should be a separate para. Sometimes paragraphs may be very short. But it's not that important, feel free to ignore my comments.

**Comment by DaanDeMeyer** [comment]
This was published so let's merge

**Pull request merged**